core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -211,6 +211,8 @@ private[deploy] class Worker(
 
   private var registerMasterFutures: Array[JFuture[_]] = null
   private var registrationRetryTimer: Option[JScheduledFuture[_]] = None
+  private[worker] var heartbeatTask: Option[JScheduledFuture[_]] = None
```

Comment:
The identifier marked as `[work]` seems to serve the purpose of merely being callable within test cases, right? Given that the current `WorkerSuite` already has `with PrivateMethodTester`, can we adopt the approach of using `invokePrivate` for testing?